src/utils/getFilteredTodos.ts

```diff
@@ -0,0 +1,36 @@
+import { Todo } from '../types/Todo';
+import { Status } from '../types/Status';
+
+type Params = {
+  query: string;
+  status: Status;
+};
+
+export const getFilteredTodos = (todos: Todo[], { query, status }: Params) => {
+  let filteredTodos = [...todos];
+
+  filteredTodos = filteredTodos.filter(todo => {
+    switch (status) {
+      case 'active':
+        return !todo.completed;
+
+      case 'completed':
+        return todo.completed;
+
+      default:
+        return todo;
```

Comment:
The default case in the switch statement should return `true` to ensure the todo is included in the filtered results. Returning the todo object itself may lead to unexpected behavior.